eppo_client/rules.py

```diff
@@ -24,25 +24,19 @@ class Condition(SdkBaseModel):
 
 
 class Rule(SdkBaseModel):
-    allocation_key: str
     conditions: List[Condition]
 
 
-def find_matching_rule(subject_attributes: dict, rules: List[Rule]):
-    for rule in rules:
-        if matches_rule(subject_attributes, rule):
-            return rule
-    return None
+def matches_rule(rule: Rule, subject_attributes: Dict[str, str]) -> bool:
+    return all(
+        evaluate_condition(condition, subject_attributes)
+        for condition in rule.conditions
+    )
 
 
-def matches_rule(subject_attributes: dict, rule: Rule):
-    for condition in rule.conditions:
-        if not evaluate_condition(subject_attributes, condition):
-            return False
-    return True
-
-
-def evaluate_condition(subject_attributes: dict, condition: Condition) -> bool:
+def evaluate_condition(
+    condition: Condition, subject_attributes: Dict[str, str]
```

Comment:
The type of `subject_attributes` now doesn't match the one in `FlagEvaluation`. Safer and more concise to have a Attributes type defined somewhere that is then shared.